pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/TableRebalancer.java

```diff
@@ -110,6 +115,8 @@ public RebalanceResult rebalance(TableConfig tableConfig, RebalanceSegmentStrate
           strategy.getRebalancedIdealState(previousIdealState, tableConfig, rebalanceConfig, partitionAssignment);
       result.setIdealStateMapping(idealState.getRecord().getMapFields());
       result.setPartitionAssignment(partitionAssignment);
+      result.setStatus(RebalanceResult.RebalanceStatus.DONE);
+      _rebalancerStats.dryRun = 1;
```

Comment:
For a given invocation of Rebalancer tool (either through command line or rest api), we create a single instance of TableRebalancer and invoke the rebalance() method. The stats are computed in a cumulative manner for that invocation. However, if the user invoked it in dry run mode then dryRun is set to 1 -- it can only be 1 for that invocation

Reply:
my bad, sorry